### PR TITLE
Fix spelling errors and update nginx docs

### LINKED ISSUE #539: Redirection parameter not changed in docs for v4.0
See https://github.com/authelia/authelia/commit/9ae2096d2a0840dc0a14925f8b1a9a7c3af715c2#diff-7e66b042bbf0a03a3f36e62386beba98L62 

PR #538  opened

### PR DESCRIPTION
**Pull request by nightah**
Closes #539.

### CODE REVIEW

File: BREAKING.md [hunk 1]
```diff
@@ -21,8 +21,8 @@ fortunately migration tools are provided to ease the task.
 * The configuration mostly remained the same, only one major key has been added: `jwt_secret`
 and one key removed: `secure` from the SMTP notifier as the Go SMTP library default to TLS
 if available.
-* The Hash router has been replaced by a Browser router. This means that the weird characters
-/%23/ in the URL could now be safely removed.
+* The Hash router has been removed and replaced with a Browser router. This means that the weird characters
```
Comment (clems4ever) [inline BREAKING.md]: This is already merged in @SilverBut 's PR, you should probably rebase.

File: cmd/authelia-scripts/migration_local.go [hunk 2]
```diff
@@ -102,7 +102,7 @@ func migrateLocalU2FSecret(dbProvider storage.Provider) {
 }
 
 func migrateLocalPreferences(dbProvider storage.Provider) {
-	file, err := os.Open(path.Join(localDatabasePath, "prefered_2fa_method"))
+	file, err := os.Open(path.Join(localDatabasePath, "preferred_2fa_method"))
```
Comment (clems4ever) [inline cmd/authelia-scripts/migration_local.go]: This change will break the migration script.

File: cmd/authelia-scripts/migration_mongo.go [hunk 3]
```diff
@@ -124,7 +124,7 @@ func migrateMongoTOTPDevices(db *mongo.Database, dbProvider storage.Provider) {
 }
 
 func migrateMongoPreferences(db *mongo.Database, dbProvider storage.Provider) {
-	u2fCollection := db.Collection("prefered_2fa_method")
+	u2fCollection := db.Collection("preferred_2fa_method")
```
Comment (clems4ever) [inline cmd/authelia-scripts/migration_mongo.go]: This change will break the migration script.

File: BREAKING.md (hunk 1 above)
Comment (nightah) [reply BREAKING.md]: I merged the PR into this branch and not master, hence why you're seeing it here.
I actually rebased it into this PR.

File: cmd/authelia-scripts/migration_local.go (hunk 2 above)
Comment (nightah) [reply cmd/authelia-scripts/migration_local.go]: Ah so the original store was misspelled?
I'll revert this and the mongo one too.

**Pull request merged**